aws/resource_aws_appautoscaling_policy.go

```diff
@@ -285,10 +286,24 @@ func resourceAwsAppautoscalingPolicyCreate(d *schema.ResourceData, meta interfac
 }
 
 func resourceAwsAppautoscalingPolicyRead(d *schema.ResourceData, meta interface{}) error {
-	p, err := getAwsAppautoscalingPolicy(d, meta)
+	var p *applicationautoscaling.ScalingPolicy
+
+	err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		var err error
+		p, err = getAwsAppautoscalingPolicy(d, meta)
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok {
+				if awsErr.Code() == applicationautoscaling.ErrCodeFailedResourceAccessException {
+					return resource.RetryableError(err)
+				}
+			}
+		}
```

Comment:
We should `return resource.NonRetryableError(err)` as the last statement inside the `if err != nil` conditional so its passed up to the returned error, otherwise the outside code will not see these. I'll fix these on merge.